build-system/tasks/visual-diff.js

```diff
@@ -143,10 +139,27 @@ async function launchWebServer() {
     host: HOST,
     port: PORT,
     retries: WEBSERVER_TIMEOUT_RETRIES, // retry timeout defaults to 1 sec
-  }).on('connected', resolver).on('timeout', rejecter);
+  }).on('connected', () => {
+    return resolver(gulpServeAsync);
+  }).on('timeout', rejecter);
   return deferred;
 }
 
+/**
+ * Kill the webserver process and this own process.
+ *
+ * @param {!ChildProcess} webServerProcess the webserver process to shut down.
+ */
+async function shutdown(webServerProcess) {
+  if (webServerProcess.exitCode == null) {
+    // Explicitly exit the webserver.
+    webServerProcess.kill();
+    // The child node process has an asynchronous stdout. See #10409.
+    await sleep(100);
+  }
+  process.exit();
```

Comment:
What is this "done" that you're talking about? I saw it in tests but not in any of the gulp files

Unfortunately the process doesn't exit unless I explicitly call process.exit(), for some reason
My guess is that somehow the child processes are hanging it. I'd be happy if you can help me figure this out! I want the process to exit cleanly...

Reply:
`done` is just a callback you can pass to a gulp task (can be called anything). Look up the gulp 3.9 docs.

For the child task, you  could try sending it a kill instead of an int, and then call done in the main task. If all that fails, add a TODO.